llama-index-integrations/indices/llama-index-indices-managed-llama-cloud/llama_index/indices/managed/llama_cloud/retriever.py

```diff
@@ -1,13 +1,39 @@
 from typing import Any, List, Optional
 
-from llama_cloud import TextNodeWithScore
+from llama_cloud import TextNodeWithScore, PageScreenshotNodeWithScore
 from llama_cloud.resources.pipelines.client import OMIT, PipelineType
+from llama_cloud.client import LlamaCloud
+from llama_cloud.core import remove_none_from_dict
+from llama_cloud.core.api_error import ApiError
 
 from llama_index.core.base.base_retriever import BaseRetriever
 from llama_index.core.constants import DEFAULT_PROJECT_NAME
 from llama_index.core.ingestion.api_utils import get_aclient, get_client
-from llama_index.core.schema import NodeWithScore, QueryBundle, TextNode
+from llama_index.core.schema import NodeWithScore, QueryBundle, TextNode, ImageNode
 from llama_index.core.vector_stores.types import MetadataFilters
+import urllib.parse
+import base64
+
+
+def _get_page_screenshot(
```

Comment:
Lets also make an async version of this, otherwise `aretrieve()` will block on this call

Reply:
good point